### PR TITLE
Frontend Directory added. 

### PR DESCRIPTION
**Pull request by OjasviSingh**
Added all the subfolders of the frontend directory.

### CODE REVIEW

**Comment by ben-AI-cybersec** [comment]
This looks good to me, but I'm not that conversant with React. @doubleedesign would you have a quick look and see if the frontend dir looks reasonable?

**Comment by doubleedesign** [comment]
I haven't been able to take a close look at this yet, but it occurs to me that if the power curve functionality is fairly complex and potentially even something that could be used by other projects (smartbike?) maybe this could be packaged up as its own thing that can be imported and used within the Project 3 (and other) apps, much like Redback UI will be.

One nitpick, not urgent to change, but going forward I really recommend using [Vite](https://vitejs.dev/) to bootstrap React apps. Create-react-app has been removed from the official docs and is broadly considered deprecated, with Vite as a go-to replacement for simple apps. (The official docs don't say to use Vite and IMO over-complicate things for simple apps by telling everyone to use Next or Remix in the first instance which is overkill for a lot of apps, so I can see how this advice could seem a little contradictory. I'd be open to using Next, Remix etc if we have a strong use case, but I digress. TL;DR CRA = old news.)

File: power-curve-react/backend/models/__pycache__/power_curve.cpython-312.pyc
Comment (ben-AI-cybersec) [inline power-curve-react/backend/models/__pycache__/power_curve.cpython-312.pyc:1]: Please delete this temp file and/or add `__pycache__` to your .gitignore

**Review by ben-AI-cybersec — APPROVED** [review]
Tested locally, works great.

**Comment by ben-AI-cybersec** [comment]
@cybermehak I think you wanted to test this for injection attacks before I merge?

**Review by ben-AI-cybersec — APPROVED** [review]
approving to remove the frontend submodule

**Pull request merged**